internal/controllers/operator_controller.go

```diff
@@ -44,11 +45,18 @@ import (
 	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
 )
 
+// BundleProvider provides the way to retrieve a list of Bundles from a source,
+// generally from a catalog client of some kind.
+type BundleProvider interface {
+	Bundles(ctx context.Context) ([]*catalogmetadata.Bundle, error)
+}
+
 // OperatorReconciler reconciles a Operator object
 type OperatorReconciler struct {
 	client.Client
-	Scheme   *runtime.Scheme
-	Resolver *solver.DeppySolver
+	CatalogClient BundleProvider
```

Comment:
I renamed the field to be `BundleProvider` to match `BundleProvider` interface. But I kept catalog client names in the client itself (at least for now). The client already fetches all the data from catalogs - It is just that we only provide a method to get bundles. This might change at some point.